VimL/confirm(): Show dialog even if :silent

justinmk opened the pull request:
closes #8788

justinmk committed:
VimL/confirm(): Show dialog even if :silent

closes #8788
related #9034

bfredl commented on test/functional/eval/input_spec.lua:
```diff
@@ -439,3 +440,40 @@ describe('inputdialog()', function()
+    check_and_clear(':silent edit foo         |\n')
+
+    -- With API (via eval/VimL) call and shortmess+=F
+    feed(':call nvim_command("edit x")<cr>')
```
What about just `command('edit x')` (rpc api), will it work the same? Regardless it could be added to the test.

justinmk replied:
Not sure how to test that since it will block, currently.

justinmk replied:
Oh, could send it as a notification.

justinmk committed:
doc: deprecate inputdialog()

input() is functionally equivalent. GUI support for dialogs is implicit
and does not depend on choosing inputdialog() vs input().
